src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+            }
+
+            Log($"Firmata version on board is {FirmataVersion}.");
+
+            FirmwareVersion = _firmata.QueryFirmwareVersion(out var firmwareName);
+            FirmwareName = firmwareName;
+
+            Log($"Firmware version on board is {FirmwareVersion}");
+
+            _firmata.QueryCapabilities();
+
+            _supportedPinConfigurations = _firmata.PinConfigurations; // Clone reference
+
+            Log("Device capabilities: ");
+            foreach (var pin in _supportedPinConfigurations)
+            {
+                Log(pin.ToString());
+            }
+
+            _firmata.EnableDigitalReporting();
+        }
+
+        /// <summary>
+        /// Firmware version on the device
+        /// </summary>
+        public Version FirmwareVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Name of the firmware.
+        /// </summary>
+        public string FirmwareName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Firmata version found on the board.
+        /// </summary>
+        public Version FirmataVersion
+        {
+            get;
+            private set;
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata ?? throw new ObjectDisposedException(nameof(ArduinoBoard));
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of capabilities per pin
+        /// </summary>
+        public ReadOnlyCollection<SupportedPinConfiguration> SupportedPinConfigurations
+        {
+            get
+            {
+                return _supportedPinConfigurations.AsReadOnly();
+            }
+        }
+
+        internal void Log(string message)
+        {
+            LogMessages?.Invoke(message, null);
+        }
+
+        private void FirmataOnError(string message, Exception? innerException)
+        {
+            LogMessages?.Invoke(message, innerException);
+        }
+
+        /// <summary>
+        /// Creates a GPIO Controller instance for the board. This allows working with digital input/output pins.
+        /// </summary>
+        /// <returns>An instance of GpioController, using an Arduino-Enabled driver</returns>
+        public GpioController CreateGpioController()
+        {
+            return CreateGpioController(DefaultPinNumberingScheme);
+        }
+
+        /// <summary>
+        /// Creates a GPIO Controller instance for the board. This allows working with digital input/output pins.
+        /// </summary>
+        /// <param name="pinNumberingScheme">Pin numbering scheme to use for this controller</param>
+        /// <returns>An instance of GpioController, using an Arduino-Enabled driver</returns>
+        public virtual GpioController CreateGpioController(PinNumberingScheme pinNumberingScheme)
```

Comment:
I thought you said that only logical pin numbering was supported by firmata driver right?

Reply:
True, but this is supposed to be overriding the generic method from the (future) base class.

Reply:
I see, in that case I would remove it from this PR since you are going to change the signature anyway when you add the board class as you will need to change virtual => override. I would try to break any ties that this PR has with the Board one for now since the Board one can still change and we don't want to add dependencies here that we may change later.

Reply:
I can do that now, but as I already said: I do expect that there will be changes for these adjustments anyway.

Reply:
I'll leave it up to you, but I think that the way to go about it is to start with the minimal, and then add the rest once the board PR comes in.

Reply:
Removed the overload with an argument for now.